lib/Log/Any/Adapter/DERIV.pm

```diff
@@ -444,50 +506,67 @@ Return : true or false
 # Using fcntl to lock a file can avoid this problem
 sub _flock {
     my ($fh, $type) = @_;
-    my $lock = _linux_flock_data($type);
+    my $lock   = _linux_flock_data($type);
     my $result = fcntl($fh, F_SETLKW, $lock);
+
     return $result if $result;
+
     print STDERR "F_SETLKW @_: $!\n";
+
     return undef;
 }
+
 =head2 _lock
 
 Lock a file handler with fcntl.
 
-Param: fh - File handle
+=over 4
 
-Return: true or false
+=item * C<$fh> File handle
+
+=back
+
+Returns boolean
 
 =cut
 
-sub _lock{
+sub _lock {
     my ($fh) = @_;
+
     return _flock($fh, F_WRLCK);
 }
 
 =head2 _unlock
 
 Unlock a file handler locked by fcntl
 
-Param: fh - File handle
-Return: true or false
+=over 4
+
+=item * C<$fh> File handle
+
+=back
+
+Returns boolean
 
 =cut
 
-sub _unlock{
+sub _unlock {
     my ($fh) = @_;
+
     return _flock($fh, F_UNLCK);
 }
+
 =head2 level
 
 return the current log level name
 
 =cut
 
 sub level {
-    my $self = shift;
-    my @methods = reverse logging_methods();
-    my %num_to_name = map {$_ => $methods[$_]} 0..$#methods;
+    my $self        = shift;
+    my @methods     = reverse logging_methods();
```

Comment:
Why are we building these data structures on every call? That seems very inefficient... perhaps provide these as package variables or at least switch to `state`?

Reply:
Seems this function is not used. @sasikala-binary Could you please confirm it and drop it ?

Reply:
used in binary-websocket-api, bom-myaffiliates and bom-oauth, so retaining it and changed to package variable